notifier.go

```diff
@@ -33,23 +33,33 @@ func New(rawData ...interface{}) *Notifier {
 // Notify sends an error to Bugsnag. Any rawData you pass here will be sent to
 // Bugsnag after being converted to JSON. e.g. bugsnag.SeverityError, bugsnag.Context,
 // or bugsnag.MetaData.
-func (notifier *Notifier) Notify(err error, rawData ...interface{}) (e error) {
-	config := notifier.Config
-	return notifier.NotifySync(err, config.Synchronous, rawData...)
+func (notifier *Notifier) Notify(rawData ...interface{}) (e error) {
+	return notifier.NotifySync(append(rawData, notifier.Config.Synchronous)...)
 }
 
 // NotifySync sends an error to Bugsnag. The synchronous parameter specifies
 // whether to send the report in the current context. Any rawData you pass here
 // will be sent to Bugsnag after being converted to JSON. e.g.
 // bugsnag.SeverityError,  bugsnag.Context, or bugsnag.MetaData.
-func (notifier *Notifier) NotifySync(err error, synchronous bool, rawData ...interface{}) (e error) {
-	event, config := newEvent(errors.New(err, 1), rawData, notifier)
+func (notifier *Notifier) NotifySync(rawData ...interface{}) (e error) {
+	containsError := false
+	for _, datum := range rawData {
+		if _, ok := datum.(error); ok {
+			containsError = true
+		}
+	}
+	if !containsError {
+		msg := "bugsnag.Notify was called without supplying an error. Bugsnag not notified"
```

Comment:
This should probably read `bugsnag.NotifySync`.

Reply:
I'm going to leave the method name out entirely - `attempted to notify Bugsnag...`